src/errors.exceptions.md

```diff
@@ -4,8 +4,18 @@ In general, prefer [explicit error handling mechanisms](errors.result.md).
 
 Annotate each module at top-level (before imports):
 
-* `{.push raises: [Defect].}` (nim 1.2+)
-* `{.push raises: [].}` (nim 1.6+)
+* `{.push raises: [Defect].}` (nim 1.2)
+* `{.push raises: [].}` (nim 1.4+)
+
+To make a module compatible with both Nim 1.2 and newer versions, use:
+
+```nim
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
```

Comment:
an optional style for this might be:

```
when (NimMajor, NimMinor) < (1, 4):  {.push raises: [Defect].}
else: {.push raises: [].}
```

Reply:
`upraises` uses a lot of complex stuff (different syntax for pushing, macros etc) for what ultimately is a trivial problem - we've avoided it in most projects for this reason (it's only codex/derivatives that use it) - it's also not quite in line with some of the longer term planning that we have revolving around an `error` pragma, so introducing the complexity has always felt meh.